src/v/raft/state_machine.cc

```diff
@@ -124,17 +128,29 @@ ss::future<result<replicate_result>> state_machine::quorum_write_empty_batch(
           });
       });
 }
-
+ss::future<> state_machine::maybe_apply_raft_snapshot() {
+    // a loop here is required as install snapshot request may be processed by
+    // Raft while handling the other snapshot. In this case a new snapshot
+    // should be applied to the STM.
+
+    while (_next < _raft->start_offset()) {
+        try {
+            co_await handle_raft_snapshot();
+        } catch (...) {
+            const auto& e = std::current_exception();
+            if (!ssx::is_shutdown_exception(e)) {
+                vlog(_log.error, "Error applying Raft snapshot - {}", e);
```

Comment:
Do we even need this log statement? It seems that this will also log in `apply` but I guess it gives us the extra information about the error was for the Raft snapshot not something else.

Reply:
actually it may be a relevant information point us to the source of the problem, f.e. the problem may lay in corrupted snapshot